packages/flutter/lib/src/widgets/viewport.dart

```diff
@@ -219,26 +222,64 @@ class _ViewportElement extends MultiChildRenderObjectElement {
 
   @override
   void mount(Element? parent, Object? newSlot) {
+    assert(!_doingMountOrUpdate);
+    _doingMountOrUpdate = true;
     super.mount(parent, newSlot);
     _updateCenter();
+    assert(_doingMountOrUpdate);
+    _doingMountOrUpdate = false;
   }
 
   @override
   void update(MultiChildRenderObjectWidget newWidget) {
+    assert(!_doingMountOrUpdate);
+    _doingMountOrUpdate = true;
     super.update(newWidget);
     _updateCenter();
+    assert(_doingMountOrUpdate);
+    _doingMountOrUpdate = false;
   }
 
   void _updateCenter() {
     // TODO(ianh): cache the keys to make this faster
     if (widget.center != null) {
-      renderObject.center = children.singleWhere(
-        (Element element) => element.widget.key == widget.center,
-      ).renderObject as RenderSliver?;
+      for (int i = 0; i < children.length; i++) {
+        if (children.elementAt(i).widget.key == widget.center) {
```

Comment:
elementAt on an Iterable is fairly inefficient as it internally iterates over the collection until it has arrived at the desired index (see implementation at: https://master-api.flutter.dev/flutter/dart-core/Iterable/elementAt.html). We should probably make this a little more efficient here by having the for loop iterate over the children and keeping a separate count variable that we increase with each iteration.